README.md

```diff
@@ -1,59 +1,100 @@
-the Imagebuilder of OpenWrt).
+The Makefile automates the build of the firmware (applies patches, integrates custom packages and uses the
+the ImageBuilder of OpenWrt).
 
 The idea is to download OpenWrt via git, patch it with all patches in the `patches/` folder
-and configure it according to the configuration snippets in `configs/`. Then use the OpenWrt
-buildsystem to build all packages and the Imagebuilder. The packages to assemble into the final
-image for the router are taken from the imageflavors defined in `packagelist/`.
+patch it and configure it according to the configuration snippets in `configs/`. Then use the OpenWrt
+build system to build all packages and the image builder. The packages that will be assembled to the final image for the router
+image for the router are taken from the image flavors defined in `packagelist/`.
 
-### Build Prerequisites
+### Build prerequisites
 
 Please take a look at the [OpenWrt documentation](https://openwrt.org/docs/guide-developer/build-system/install-buildsystem?s[]=prerequisites#prerequisites)
-for a complete and uptodate list of packages for your operating system. 
+for a complete and up-to-date list of packages for your operating system. 
 
-On Ubuntu/Debian:
+Alpine:
 ```
-apt-get install git build-essential libncurses5-dev zlib1g-dev gawk time \
-  unzip libxml-perl flex wget gawk gettext quilt python libssl-dev
+apk add asciidoc bash bc binutils bzip2 cdrkit coreutils diffutils \
+findutils flex g++ gawk gcc gettext git grep intltool libxslt \
+linux-headers make ncurses-dev openssl-dev patch perl \
+python2-dev python3-dev rsync tar unzip util-linux wget zlib-dev
 ```
 
-On openSUSE:
+Arch / Manjaro:
+```
+# Essential prerequisites
+pacman -S --needed base-devel bash bzip2 git libelf libxslt ncurses \
+openssl python2 time unzip util-linux wget zlib
+ 
+# Optional prerequisites, depend on the package selection
+pacman -S --needed asciidoc help2man intltool perl-extutils-makemaker
+```
+
+CentOS / Fedora:
+```
+sudo dnf --skip-broken install bash-completion bzip2 gcc gcc-c++ git \
+make ncurses-devel patch perl-Data-Dumper perl-Thread-Queue python2 \
+python3 rsync tar unzip wget perl-base perl-File-Compare \
+perl-File-Copy perl-FindBin diffutils which
+```
+
+Debian / Ubuntu:
+```
+sudo apt install build-essential ccache ecj fastjar file g++ gawk \
+gettext git java-propose-classpath libelf-dev libncurses5-dev \
+libncursesw5-dev libssl-dev python python2.7-dev python3 unzip wget \
+python3-distutils python3-setuptools python3-dev rsync subversion swig time \
+xsltproc zlib1g-dev 
 ```
-zypper install --type pattern devel_basis
-zypper install git ncurses-devel zlib-devel gawk time \
-  unzip perl-libxml-perl flex wget gawk gettext-runtime quilt python libopenssl-devel
+
+Gentoo:
+```
+echo \
+app-arch/{bzip2,sharutils,unzip,zip} sys-process/time \
+app-text/asciidoc \
+dev-libs/{libusb-compat,libxslt,openssl} dev-util/intltool \
+dev-vcs/{git,mercurial} net-misc/{rsync,wget} \
```

Comment:
mercurial / hg + + subversion(debian listing) are not required on any build I did in the last months

Reply:
I've used the packages listed by OpenWrt (https://openwrt.org/docs/guide-developer/build-system/install-buildsystem#examples_of_package_installations)

Reply:
Looks like the wiki-page mentions also prerequisites for building exotic packages. I would avoid to "blindly" install Java, svn, hg as we do not need them to build our firmware, and it "bloats" the list of packages only.  The OpenWrt-list looks quite inconsistent at all, just compare "Arch" and "Debian".
If the user start to compile exotic packages, it might be handy to refreence the OpenWrt-Wikipage for additional reading. But this I would not consider "regular" usecase.

Reply:
Okay, I will change that. I thought that since OpenWrt even suggests these dependencies, it would apply here as well. But since we avoid building these exotic packages, we don't need so many make dependencies.

Reply:
looking forward to it

Reply:
The packages have been updated. You can name packages that are not yet needed to build the firmware and I will remove them from the list